blog/2023-11-01-feature-flags-sdks-architectures.md

```diff
@@ -0,0 +1,127 @@
+Feature flag evaluation is done locally via the SDK and not affecting network traffic.
+With this approach, configuration can be fetched on initialization, and be refreshed periodically and/or when there is a configuration change, which can be triggered at the SDK via stream event from the feature flag service. 
+Some SDKs sync data via WebSockets or [SSE](https://en.wikipedia.org/wiki/Server-sent_events).
+When this approach is used, evaluation analytics metrics can be aggregated at the SDK side and published separately, as not every flag evaluation triggers a message to the Features Service / Proxy.
+
+### Local evaluation - example flow
+
+<img src={require('@site/static/img/blog/feature-flags-sdks-architectures/Local-evaluation.png').default} />
+
+#### Advantages
+
+* Functional advantage is that intermediate issue like network or feature flag service errors are automatically handled, as configuration is cached, and flags are evaluated locally.
+  If an issue like network error is ongoing for a long time period, SDK provider state can also be reflected as stale.
+* Another functional advantage, initialization can have a fallback mechanism to dealing with situations like temporary network failure by starting from last received configuration or some other configuration, thus microservice can start with last feature flag configuration.
+* Feature flag configuration can be cached on CDN.
+* Feature flag evaluation is done locally via the SDK and not affecting network traffic, hence can be fast and efficient.
+* Working without a relay proxy can still be efficient from SDK point of view, as flags are evaluated locally and network traffic does not block common flow.
+
+#### Disadvantages
+
+* When feature flag configuration is very large with many feature flags, network traffic and local cache sizes can be increased.
+* Local evaluation should be implemented at the SDK, developed and maintained for multiple languages. Considering rules like context properties based and percentage rollout strategies via hashing to buckets, it can be hard to keep 100% feature parity across languages.
+
+#### A note on client-side SDKs for web browser / client app
+
+This approach is less suitable for client-side SDKs from a security point of view, as full feature flag configuration and authorization keys are exposed to end users.
+
+## OpenFeature and server-side SDKs
+
+* OpenFeature has the [specification](https://openfeature.dev/specification/glossary/#feature-flag-sdk) for SDKs for
+  multiple programming languages for developing a unified API and SDK.
+* The OpenFeature specification has methods like flags evaluation and initialization.
+* The OpenFeature specification treats events, which can go together with some scenarios like network errors.
+* As there are multiple SDK architecture approaches, architectures and fallback mechanisms are not always
+  well-defined, and vendors can implement providers the way they choose.
+
+
```

Comment:
Could you add a short call to action section? Something about creating a provider and joining the OpenFeature community.

Reply:
mm not sure it would fit to the post for my point of view.. is that ok ?

Reply:
Sure, it's up to you.